tests/C-tests/CMakeLists.txt

```diff
@@ -109,41 +108,49 @@ function(test_formats ${formats})
   endforeach()
 endfunction(test_formats)
 
-if(BML_MPI)
-  set(testlist
-      allocate)
-else()
-  set(testlist
-    add
-    adjacency
-    adjungate_triangle
+function(test_formats_mpi ${formats})
+  foreach(T ${formats} )
+    foreach(P single_real double_real single_complex double_complex)
+      add_test(MPI-C-${N}-${T}-${P} mpirun -np 4 ./bml-test -n ${N} -t ${T} -p ${P})
```

Comment:
For some reason, `mpirun` is very sensitive to the path in the executable. When you add `./bml-test` with the relative path it starts working.